### DISCUSSION TITLE
Is there any difference between equality constraint and assignment

### DISCUSSION THREAD

**Discussion by Geriro36** [yalmip/YALMIP#940] [Optimization modelling] [answered]
Hi,Johan
I'm trying to do some case studies. When some variables have been fixed, I'm not sure whether to use assignment or equality constraint.
For example，in this model， if I know that x (2) and X (5) are always 0.
`x=sdpvar(1,6);
A=rand(6,6);
C=[ ];
C=[0<=A * x<=2]; %
`
 Are there any differences between the following two expressions？
`  x（[ 2 ,5]）=0`   or   `C=[C   x（[ 2 ,5]）==0 ]`
In addition，
I find that sometimes binary variables are not absolutely equal to 0 after optimization.
If it is a binary variable, is the optimization results different in some other more complex models?

Thanks and Regards,

Xin

**Comment by johanlofberg** [comment, marked as answer]
This is an optimization problem in 6 variables
```
x=sdpvar(1,6); 
A=rand(6,6);
C=[0<=A * x<=2]; 
C=[C x（[ 2 ,5]）==0 ]
optimize(C)
```

This is an optimization problem in 4 variables
```
x=sdpvar(1,6); 
x（[ 2 ,5]）=0
A=rand(6,6);
C=[0<=A * x<=2]; 
optimize(C)

```
and is a convenient way of writing
```
x=[sdpvar(1) 0 sdpvar(1) sdpvar(1) 0 sdpvar(1)]
A=rand(6,6);
C=[0<=A * x<=2]; 
optimize(C)
```
Some solver might be more efficient with equalities in high dimensions, some inefficient and prefer a condensed form, and most likely it makes no difference as the solver will presolve the model anyway to put it in its preferred form

Solvers only solve problems up to specified numerical tolerance, and might thus consider 0.999999999 to be integer enough. Some solvers will post-process and clean up integers, most won't.

**Reply by Geriro36** [reply]
Thank you for your answer.
